src/atomate2/classical_md/openmm/jobs/base.py

```diff
@@ -0,0 +1,494 @@
+    tags: list[str] | None = field(default=None)
+    n_steps: int | None = field(default=None)
+    step_size: float | None = field(default=None)
+    temperature: float | None = field(default=None)
+    friction_coefficient: float | None = field(default=None)
+    platform_name: str | None = field(default=None)
+    platform_properties: dict | None = field(default=None)
+    state_interval: int | None = field(default=None)
+    state_file_name: str | None = field(default=None)
+    traj_interval: int | None = field(default=None)
+    wrap_traj: bool | None = field(default=None)
+    report_velocities: bool | None = field(default=None)
+    traj_file_name: str | None = field(default=None)
+    traj_file_type: str | None = field(default=None)
+    embed_traj: bool | None = field(default=None)
+
+    @openff_job
+    def make(
+        self,
+        interchange: Interchange | bytes,
+        prev_task: OpenMMTaskDocument | None = None,
+        output_dir: str | Path | None = None,
+    ) -> Response:
+        """Run an OpenMM calculation.
+
+        This method sets up an OpenMM simulation, runs the simulation based
+        on the specific job logic defined in run_openmm, and closes the
+        simulation. It returns a response containing the output task document.
+
+        Parameters
+        ----------
+        interchange : Union[Interchange, str]
+            An Interchange object or a JSON string of the Interchange object.
+        prev_task : Optional[OpenMMTaskDocument]
+            The previous task document.
+        output_dir : Optional[Union[str, Path]]
+            The directory to save the output files.
+            Resolution order is output_dir > prev_task.dir_name > Path.cwd(). Will
+            create a directory if needed.
+
+        Returns
+        -------
+        Response
+            A response object containing the output task document.
+        """
+        # this is needed because interchange is currently using pydantic.v1
+        if isinstance(interchange, bytes):
+            interchange = interchange.decode("utf-8")
+        if isinstance(interchange, str):
+            interchange = Interchange.parse_raw(interchange)
+        else:
+            interchange = copy.deepcopy(interchange)
+
+        dir_name = Path(
+            output_dir or getattr(prev_task, "dir_name", None) or Path.cwd()
+        )
+        dir_name.mkdir(exist_ok=True, parents=True)
+
+        sim = self._create_simulation(interchange, prev_task)
+
+        self._add_reporters(sim, dir_name, prev_task)
+
+        # Run the simulation
+        start = time.time()
+        self.run_openmm(sim)
+        elapsed_time = time.time() - start
+
+        self._update_interchange(interchange, sim, prev_task)
+
+        del sim
+
+        task_doc = self._create_task_doc(interchange, elapsed_time, dir_name, prev_task)
+
+        # write out task_doc json to output dir
+        with open(dir_name / "taskdoc.json", "w") as file:
```

Comment:
Why do we write the task doc? I'm not entirely opposed to this but it will get stored in the database anyway.

Reply:
Mainly to enable easier local analysis.